### DISCUSSION TITLE
node_exporter의 데이터를 외부 서버의 grafana에서 끌고 오지 못하는 문제

### DISCUSSION THREAD

**Discussion by blahxblah** [hanul-pipeline/stopped_hanul-site-pipeline#13] [Q&A] [answered]
도커로 실행된 node exporter의 데이터가 외부 서버의 grafana에서 표현이 안됩니다.
prometheus에서 node exporter 서버의 외부 ip 설정을 잡아주었는데 데이터를 끌고 오지 못하는 것 같아요.
도커서버 상에서는 node exporter metrics는 제대로 실행되고 있는데 네트워크에 문제가 있어 보입니다.
네트워크 설정 관련해서 같이 고민을 해봐야 할 것 같아요.

**Comment by hooniegit** [comment, marked as answer]
아 그거 제가 테스트를 해 봤었는데 아마 네트워크 라우팅 관련 문제 같습니다.
우선 네트워크 통신이 가능한 상태여야 하기 때문에 두개의 서버가 같은 네트워크망에 연결되어 있어야 하고,
해당 네트워크의 라우터 설정이 '상호 통신이 가능하도록' 구성되어야 하는 것 같아요.

일례로 컴퓨터 2대로 스타벅스 공용 인터넷으로 진행했을 때는 grafana도 안 되고 curl 커맨드 전송도 되질 않았는데,
개인용 핫스팟을 통해 연결하니 둘 다 정상적으로 해결이 되었습니다.

따라서 지금 저희가 크게 고민해야 할 문제는 아니지만,
실제 산업 현장의 규모가 큰 만큼, 모든 장비들을 동일 네트워크로 연결할 수 없는 문제가 발생한다면 해결 방안이 필요할 것 같습니다.
다음 주 모임 내로 해당 이슈도 이야기하면 좋을 것 같아요!
